random/README.md

```diff
@@ -0,0 +1,181 @@
+attest simple function call that generates random numbers to simulate a die roll.
+
+Before starting this example, make sure you are familiar with the
+[Hello World - Attesting a Function Call](../hello_world_attest_fn_call/README.md)
+and the
+[Error Handling](../error_handling/README.md)
+examples.
+
+You'll learn how to:
+
+- Generate random numbers in the Blocky AS runtime
+- Create a function that returns a random number within a given range
+
+*Note: Random number generation only works on Blocky AS versions 0.1.0-beta.5
+and above. While this example may run on previous versions it will generate
+deterministic number streams.*
+
+## Setup
+
+- Install the Blocky AS CLI by following the
+  [setup instructions](https://blocky-docs.redocly.app/attestation-service/setup)
+  in the Blocky AS documentation.
+- Make sure you also have
+  [Docker](https://www.docker.com/) and [jq](https://jqlang.org/) installed on
+  your system.
+
+## Quick Start
+
+To run this example, call:
+
+```bash
+make run
+```
+
+You will see the following output extracted from a Blocky AS response:
+
+```
+Output:
+{
+  "Success": true,
+  "Error": "",
+  "Value": 3
+}
+```
+
+## Walkthrough
+
+### Step 1: Create a parameterized random function
+
+We'll demonstrate random by emulating a die roll in `rollDie` in
+[`main.go`](./main.go). As in previous examples, we will call this function
+using the `bky-as` CLI by passing in the
+[`fn-call.json`](./fn-call.json) file contents:
+
+```json
+{
+  "code_file": "tmp/x.wasm",
+  "function": "rollDie",
+  "input": {
+    "sides": 20
+  }
+}
+```
+
+Notice the `input` section, which contains the parameters for `rollDie`,
+specifically the `sides` field set to `20`. Meaning, `rollDie` will return
+a number in the range [1,20].
+
+Next, we define the `rollDie` function in [`main.go`](./main.go):
+
+```go
+type Args struct {
+    Sides uint64 `json:"sides"`
+}
+
+//export rollDie
+func rollDie(inputPtr uint64, secretPtr uint64) uint64 {
+    var input Args
+    inputData := basm.ReadFromHost(inputPtr)
+    err := json.Unmarshal(inputData, &input)
+    switch {
+    case err != nil:
+        outErr := fmt.Errorf("could not unmarshal input args: %w", err)
+        return WriteError(outErr)
+    case input.Sides == 0:
+        outErr := fmt.Errorf("die cannot have zero sides")
+        return WriteError(outErr)
+    }
+
+    randBytes := make([]byte, 8)
+    n, err := crand.Read(randBytes)
+    switch {
+    case err != nil:
+        outErr := fmt.Errorf("reading random bytes: %w", err)
+        return WriteError(outErr)
+    case n != 8:
+        outErr := fmt.Errorf("reading random bytes: expected 8 bytes, got %d", n)
+        return WriteError(outErr)
+    }
+
+    roll := (binary.BigEndian.Uint64(randBytes) % input.Sides) + 1
+    return WriteOutput(roll)
+}
+```
+
+You will notice a few things:
+
+- The `rollDie` function is exported so that it can be invoked by the
```

Comment:
I think we can take this out - already covered by the hello-world example

Reply:
Removed